src/Facade/Services/ConsignmentShipfileFacadeService.cs

```diff
@@ -35,25 +36,27 @@ public IResult<IEnumerable<ConsignmentShipfile>> GetShipfiles()
         public IResult<IEnumerable<ConsignmentShipfile>> SendEmails(
            ConsignmentShipfilesSendEmailsRequestResource toSend)
         {
-            try
-            {
-                var result = this.domainService.SendEmails(
-                    toSend.Shipfiles.Select(
-                        s => new ConsignmentShipfile { Id = s.Id, ConsignmentId = s.ConsignmentId }),
-                    toSend.Test,
-                    toSend.TestEmailAddress);
-
-                if (!toSend.Test)
+            var result = new List<ConsignmentShipfile>();
+           
+                foreach (var resource in toSend.Shipfiles)
                 {
-                    this.transactionManager.Commit();
-                }
+                    try
+                    {
+                        var sent = this.domainService.SendEmails(
+                            new ConsignmentShipfile { Id = resource.Id, ConsignmentId = resource.ConsignmentId },
+                            toSend.Test,
+                            toSend.TestEmailAddress);
+                        result.Add(sent);
 
+                        this.transactionManager.Commit();
+                    }
+                    catch (PdfServiceException exception)
+                    {
+                        return new ServerFailureResult<IEnumerable<ConsignmentShipfile>>(exception.Message);
```

Comment:
I'm not sure of context but would it be useful for us to know which email it failed on in the message or would we just have to go and check ourselves anyway?